pkg/alertmanager/alertmanager.go

```diff
@@ -433,9 +432,9 @@ func clusterWait(position func() int, timeout time.Duration) func() time.Duratio
 
 // ApplyConfig applies a new configuration to an Alertmanager.
 func (am *Alertmanager) ApplyConfig(conf *definition.PostableApiAlertingConfig, tmpls []alertingTemplates.TemplateDefinition, rawCfg string, tmplExternalURL *url.URL, staticHeaders map[string]string) error {
-	templates := make([]io.Reader, 0, len(tmpls))
```

Comment:
The problem is not really the `io.Reader`, that's just an interface. You could also keep the type signature and create a wrapper that implements `Reader` - e.g.:

```go
type unlimitedStringReader struct {
    s string
}

func (r *unlimitedStringReader) Read(p []byte) (n int, err error) {
    return strings.NewReader(r.s).Read(p)
}
```

Reply:
Yes, but we don't need this level of abstraction.
We're currently using readers just because we need to pass readers down the pipeline, it's a relic from reading templates from disk. We can do perfectly fine using plain strings.

Reply:
> it's a relic from reading templates from disk

Ahh, tks for the context. It does make more sense that we'd like to get rid of it. I'm testing it now and should approve in a bit 👍